scalding-core/src/test/scala/com/twitter/scalding/StringUtilityTest.scala

```diff
@@ -0,0 +1,62 @@
+package com.twitter.scalding
+
+import org.scalatest.{ Matchers, WordSpec }
+
+class StringUtilityTest extends WordSpec with Matchers {
+  "fastSplitTest" should {
+    "be able to split white space" in {
+      val text1 = "this is good time"
+      val res1 = StringUtility.fastSplit(text1, " ") // split single white space
+      res1 should be {
+        Seq("this", "is", "good", "time")
+      }
+    }
+  }
+  "be able to split other separators" in {
+    val text2 = "a:b:c:d:"
+    val res2 = StringUtility.fastSplit(text2, ":")
+    res2 should be {
+      Seq("a", "b", "c", "d")
+    }
+  }
+  "be able to split only one separators" in {
+    val text2 = "a@"
+    val res2 = StringUtility.fastSplit(text2, "@")
+    res2 should be {
+      Seq("a")
+    }
+  }
+  "be able to split when separator doesn't show up" in {
+    val text2 = "a"
+    val res2 = StringUtility.fastSplit(text2, "@")
+    res2 should be {
+      Seq("a")
+    }
+  }
+  "be able to generate the same result with Java's split for random strings and separators" in {
```

Comment:
for this case, let's instead use a property check. It will be much easier:

```scala
property("fastSplit on comma is correct") = forAll { (s: String) =>
  s.split(",") == fastSplit(s, ",")
}
```
see some of the other property check style tests in the repo (those that extend Properties) to see an example.